dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/BundleCacheBean.java

```diff
@@ -0,0 +1,22 @@
+package edu.harvard.iq.dataverse;
+
+import javax.ejb.Singleton;
+import java.util.ResourceBundle;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * The purpose of this bean is to have one instance of cache
+ * of internal ResourceBundles for entire application.
+ * It's strongly recommended to
+ */
+@Singleton
+public class BundleCacheBean {
+    private static final ConcurrentMap<String, ResourceBundle> cache = new ConcurrentHashMap<>();
+
+    // -------------------- LOGIC --------------------
+
+    public ConcurrentMap<String, ResourceBundle> acquireCache() {
+        return cache;
```

Comment:
Yes – it looks strange.
1. We cannot extend `BundleUtil` without making all its methods non-static, which would be very cumbersome, as it's used in many places.
2. We could initialize new instance of `ConcurrentMap` in every instance of `CachedBundleUtil` – but this would impose some performance penalty on every call to `search(…)` and I'm not sure whether it's a good idea to have references to bundles in so many places (as every search would create a new map referencing them).
3. We could circumvent the problem with many instances of maps by making the bean a service, but then again we would have performance penalty from the way the service calls are done in JEE.
4. Last option is to have `static` `ConcurrentMap` inside `CachedBundleUtil` and it seems that it should work properly. Maybe that would be the way to go?

Reply:
As for `1)` I didn't mean extending BundleUtil but changing bundle util so it would have map `ConcurrentMap<String, ResourceBundle>`. Would it work that way? Or am I missing something?

Reply:
Ah, ok, I've misread the class name.

Reply:
So we can indeed extend `ResourceBundle`, but in that case:
1. Standard methods (ie. `get…`) would be unusable as we must provide info about which bundle should be used. Only `getBundle` could be properly implemented in this case, but that's the only advantage of extending the class.
2. Even then we have to store somewhere the cache (`ConcurrentMap`). We could do this as in point 4. from my first comment (ie. as `static`), but in this case I'd prefer to put it inside `CachedBundleUtil`.

Reply:
(On the other hand if we would make cache static, then in fact we could put it inside `BundleUtil`)

Reply:
The reason behind why I'm asking is because keeping two, almost identical implementations is not so pretty. If it would be possible to keep this map in BundleUtil it would be much cleaner solution.

Reply:
Ok, so I'll try to put the cache as static field inside `BundleUtil`. If it works it will also boost performance in other places.